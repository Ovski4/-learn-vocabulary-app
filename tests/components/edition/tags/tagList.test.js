import reducer from '../../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import TagList from '../../../../src/components/edition/tags/tagList/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('TagList component', () => {

    it('Should render the component', () => {

        const store = createStore(reducer, {
            tags: [],
            config: {
                locale: 'en'
            }
        });

        const rendered = renderer.create(
            <Provider store={store}>
                <TagList />
            </Provider>
        ).toJSON();
        expect(rendered).toMatchSnapshot();
    });

});
