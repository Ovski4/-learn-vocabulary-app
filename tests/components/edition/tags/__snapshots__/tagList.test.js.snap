// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList component Should render the component with the english locale 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Tags
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#bbb",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <View
          style={
            Object {
              "borderColor": "#bbb",
              "borderTopWidth": 0.4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontStyle": "italic",
                "lineHeight": 50,
              }
            }
          >
            Add new tags to categorize your translations
          </Text>
        </View>
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      index={[Function]}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "borderColor": "#bbb",
              "borderTopWidth": 0.4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontStyle": "italic",
                "lineHeight": 50,
              }
            }
          >
            Add new tags to categorize your translations
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`TagList component Should render the component with the french locale 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Étiquettes
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#bbb",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <View
          style={
            Object {
              "borderColor": "#bbb",
              "borderTopWidth": 0.4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontStyle": "italic",
                "lineHeight": 50,
              }
            }
          >
            Créez des étiquettes pour catégoriser vos traductions
          </Text>
        </View>
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      index={[Function]}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "borderColor": "#bbb",
              "borderTopWidth": 0.4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontStyle": "italic",
                "lineHeight": 50,
              }
            }
          >
            Créez des étiquettes pour catégoriser vos traductions
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
